tests/js/destructuring-types/prop-types.js

```diff
@@ -0,0 +1,14 @@
+// destructured props type all on one line (small enough for one line):
+const Component1 = (props: { first: String }) => abc()
```

Comment:
There is no such thing as "destructuring/destructured types". [Object types](https://www.typescriptlang.org/docs/handbook/2/objects.html) have nothing to do with destructuring.

It looks like you're trying to fix/change a lot of different things at once in one PR. It's a doubtful strategy.

Reply:
The tests in `prop-types.js` cover in more detail something that my changes had broken in `tests/flow/function-single-destructuring/object.js`. I would favor keeping all of the tests in `prop-types.js`, testing across all ES parsers, to avoid breaking these cases in the future. Any suggestions concerning the naming of what I am testing would be much appreciated.

At this point all of the source code changes are in `src/language-js/print/object.js`. Most of the changes in this PR are in the tests.

Reply:
I have removed these tests and reverted the update to `tests_config`. I would like to improve the testing in another PR (someday).